### PR TITLE
P3A BTC and ZEC TransactionSent

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/36791

Adding `Brave.Wallet.BtcTransactionSent` and `Brave.Wallet.ZecTransactionSent` for user approving corresponding coin transaction similar to `Brave.Wallet.EthTransactionSent` or other coins

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Needs testing similar to `Brave.Wallet.EthTransactionSent`, `Brave.Wallet.SolTransactionSent` or `Brave.Wallet.FilTransactionSent`

### CODE REVIEW

**Review by iefremov — APPROVED** [review]
p3a/ lgtm

**Review by DJAndries — APPROVED** [review]
lgtm

File: components/brave_wallet/browser/bitcoin/bitcoin_tx_manager_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,217 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (yrliou) [inline components/brave_wallet/browser/bitcoin/bitcoin_tx_manager_unittest.cc:1]: 2024

File: components/brave_wallet/browser/brave_wallet_p3a_unittest.cc [hunk 2]
```diff
@@ -633,4 +663,27 @@ TEST_F(BraveWalletP3AUnitTest, FilTransactionSentObservation) {
   histogram_tester_->ExpectUniqueSample(kFilTransactionSentHistogramName, 1, 1);
 }
 
+TEST_F(BraveWalletP3AUnitTest, BtcTransactionSentObservation) {
```
Comment (yrliou) [inline components/brave_wallet/browser/brave_wallet_p3a_unittest.cc:666]: Could we add a case for zcash too?

**Review by yrliou — APPROVED** [review]

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on

```
Brave | 1.66.19 Chromium: 123.0.6312.46 (Official Build) nightly (64-bit)
-- | --
Revision | f059d55f6082b23ef003e2f3010ebe1ea50d5b54
OS | Windows 10 Version 22H2 (Build 19045.4046)
```

**_BTCAccouunts_**

<details>
<summary>Brave.Wallet.BtcTransactionSent = 0_PASSED</summary>

1. Clean profile 1.66.x, launch browser using `brave.exe --p3a-count-wallet-test-networks`
2. Go to brave://settings/wallet/networks and click on 3 dot menu for `Bitcoin Testnet` and click on set as active
3. Create a wallet, enable the Show test networks switch while creating a wallet
4. Create a Bitcoin account in Testnet
5. Send Bitcoin to Bitcoin account via https://bitcoinfaucet.uo1.net/
6. Ensured that the Bitcoin appeared in the Bitcoin account
7. Ensure that the Transactions list is empty under `brave://wallet/crypto/portfolio/assets/0--bitcoin_testnet#transactions`
8. Open brave://local-state file and ensured that the metrics `Brave.Wallet.BtcTransactionSent` is not shown as there are no Bitcoin transaction sent

Example | Example | Example
---------|------------|--------
image | image | image

</details>

<details>
<summary>Brave.Wallet.BtcTransactionSent = 1_PASSED</summary>

1. Clean profile 1.66.x, launch browser using `brave.exe --p3a-count-wallet-test-networks`
2. Go to brave://settings/wallet/networks and click on 3 dot menu for `Bitcoin Testnet` and click on set as active
3. Create a wallet, enable the Show test networks switch while creating a wallet
4. Create a Bitcoin account in Testnet
5. Send Bitcoin to Bitcoin account via https://bitcoinfaucet.uo1.net/
6. Ensured that the Bitcoin appeared in the Bitcoin account
7. Ensure that the Transactions list is empty under `brave://wallet/crypto/accounts/0_5_0_0/transactions`
8. Open brave://local-state file and ensured that the metrics `Brave.Wallet.BtcTransactionSent` is not shown as there are no Bitcoin transaction sent
9. Create a second Bitcoin account and send a min amount to second bitcoin account
10. Ensured that the Bitcoin appeared in the Bitcoin account 2
11. Ensure that the Transactions entry is listed under Bitcoin account 1 via `brave://wallet/crypto/portfolio/assets/0--bitcoin_testnet#transactions`
12. Reload brave://local-state file and ensured that the metrics `Brave.Wallet.BtcTransactionSent` response value is `1` as expected

Example | Example | Example | Example
---------|------------|----------|---------
image | image | image | image

13. Close the browser and set the sys date ahead by 7 days
14. Relaunch the browser
15. Reload brave://local-state file and ensured that the metrics `Brave.Wallet.BtcTransactionSent` response value is `1` as expected
image
16. Close the browser and set the sys date ahead by 8 days
17. Relaunch the browser
18.  Reload brave://local-state file and ensured that the metrics `Brave.Wallet.BtcTransactionSent` response value is `0` as expected
image


</details>

**_ZECAccouunts_**

<details>
<summary>Brave.Wallet.ZecTransactionSent = 0</summary>

</details>

<details>
<summary>Brave.Wallet.ZecTransactionSent = 1</summary>

</details>

**Pull request merged**